modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -1138,23 +1138,68 @@ PolygonMeshGeneratorBase::quadElemDef(ReplicatedMesh & mesh,
 
 Real
 PolygonMeshGeneratorBase::radiusCorrectionFactor(const std::vector<Real> & azimuthal_list,
-                                                 const bool full_circle) const
+                                                 const bool full_circle,
+                                                 const unsigned int order,
+                                                 const bool is_first_value_vertex) const
 {
-  std::vector<Real> azimuthal_list_alt;
-  Real tmp_acc = 0.0;
-  Real tmp_acc_azi = 0.0;
-  azimuthal_list_alt.insert(azimuthal_list_alt.end(), azimuthal_list.begin(), azimuthal_list.end());
-  if (full_circle)
-    azimuthal_list_alt.push_back(azimuthal_list.front() + 360.0);
-  // summation of triangles S = 0.5 * r * r * Sigma_i [sin (azi_i)]
-  // Circle area S_c = pi * r_0 * r_0
-  // r = sqrt{2 * pi / Sigma_i [sin (azi_i)]} * r_0
-  for (unsigned int i = 1; i < azimuthal_list_alt.size(); i++)
+  if (order == 1)
   {
-    tmp_acc += std::sin((azimuthal_list_alt[i] - azimuthal_list_alt[i - 1]) / 180.0 * M_PI);
-    tmp_acc_azi += (azimuthal_list_alt[i] - azimuthal_list_alt[i - 1]) / 180.0 * M_PI;
+    std::vector<Real> azi_interval_list;
+    for (unsigned int i = 1; i < azimuthal_list.size(); i++)
+      azi_interval_list.push_back((azimuthal_list[i] - azimuthal_list[i - 1]) / 180.0 * M_PI);
+    if (full_circle)
+      azi_interval_list.push_back((azimuthal_list.front() + 360.0 - azimuthal_list.back()) / 180.0 *
+                                  M_PI);
+    // summation of triangles S = 0.5 * r * r * Sigma_i [sin (azi_i)]
+    // Circle area S_c = pi * r_0 * r_0
+    // r = sqrt{2 * pi / Sigma_i [sin (azi_i)]} * r_0
+    Real tmp_acc = 0.0;
+    Real tmp_acc_azi = 0.0;
+    for (const auto i : index_range(azi_interval_list))
+    {
+      tmp_acc += std::sin(azi_interval_list[i]);
+      tmp_acc_azi += azi_interval_list[i];
+    }
+    return std::sqrt(tmp_acc_azi / tmp_acc);
+  }
+  else // order == 2
```

Comment:
@roystgnr for a quad9, does the position of a middle node influence the shape of the outer boundary?

Reply:
It does not.  Not for Lagrange mappings, not for Rational Bezier Bernstein mappings.

There are mappings with less locality than the ones we use, but we'll probably never support any directly, only indirectly the way we support IsoGeometric Analysis NURBS mappings - via extraction operators that give more locally-defined elements to integrate on.

Reply:
thanks

how exact is the volume computation for second order elements in libmesh? I still see tiny diffs (1e-5) with the correction we have here

Reply:
I think that summation of small $\sin^3(\theta)$ compromises the errors...

Reply:
could be

are you summing two separate quadratic bands, separated at the mid point of the edge?
I think the whole edge is covered by a single quadratic band?

Reply:
> But are you diffing against the exact biquadratic volume, or against a perfect circle? Biquadratics (in Lagrange; I'm working on RBB mesh generator tools) aren't enough to exactly mesh a circle; no polynomial mapping is. The difference between a unit circle and its biquadratic approximation is probably more like 1e-5.

I am diffing against the corrected 1st order mesh, which should match the perfect circle volume exactly already.
While the bi-quadratics do not exactly mesh the circle, we are correcting them here so that the volume is the exact same. This is what this routine is intended for. So yes it's only correcting a very small error already

Reply:
> could be
> 
> are you summing two separate quadratic bands, separated at the mid point of the edge? I think the whole edge is covered by a single quadratic band?

Yes, it's a single band determined by the three nodes.

Reply:
If we really want to eliminate that 1e-6 relative error, one possible way would be to make a dummy TRI6 mesh with the same external nodes and use its area as the reference to derive a correction factor.

Reply:
It s more that if we expect the error to be 1e-6 already with quad-elements (as Roy mentioned), then why do we still see that after correction

I m yet to run this again and make up my mind on this

Reply:
it might just be that the error is a larger than expected pre-correction

Reply:
I will also run more cases to get a clearer idea what is going on.